File: 10.0.3.020/examples/vmlf/source/mode.f
!*******************************************************************************
!                              INTEL CONFIDENTIAL
!  Copyright(C) 2001-2008 Intel Corporation. All Rights Reserved.
!  The source code contained  or  described herein and all documents related to
!  the source code ("Material") are owned by Intel Corporation or its suppliers
!  or licensors.  Title to the  Material remains with  Intel Corporation or its
!  suppliers and licensors. The Material contains trade secrets and proprietary
!  and  confidential  information of  Intel or its suppliers and licensors. The
!  Material  is  protected  by  worldwide  copyright  and trade secret laws and
!  treaty  provisions. No part of the Material may be used, copied, reproduced,
!  modified, published, uploaded, posted, transmitted, distributed or disclosed
!  in any way without Intel's prior express written permission.
!  No license  under any  patent, copyright, trade secret or other intellectual
!  property right is granted to or conferred upon you by disclosure or delivery
!  of the Materials,  either expressly, by implication, inducement, estoppel or
!  otherwise.  Any  license  under  such  intellectual property  rights must be
!  express and approved by Intel in writing.
!
!*******************************************************************************
!  Content:
!    vmlSetMode/vmlGetMode  Example Program Text
!*******************************************************************************

      program MKL_VML_TEST

      include "mkl_vml.fi"
      include "func_interv.fi"

      integer mode
      integer(kind=4) tmode

      mode=0

      print *,"vmlSetMode/vmlGetMode example program"
      print *,""
      print *,""

      mode=VMLGETMODE()
      print 10,"Default value of vmlMode: ",mode,"h"
      call PrintTextVmlMode(mode)

      mode=jior(VML_LA,jior(VML_FLOAT_CONSISTENT,VML_ERRMODE_IGNORE))
      tmode=VMLSETMODE(mode)
      mode=VMLGETMODE()
      print 10,"Value of vmlMode after using vmlSetMode: ",mode,"h"
      call PrintTextVmlMode(mode)

10    format(A,Z,A)

      end


      subroutine PrintTextVmlMode(mode)

      include "mkl_vml.fi"

      integer(kind=4) mode
      integer(kind=4) mask

      mask=VML_ACCURACY_MASK
      if(jiand(mode,mask).eq.1) print *,"VML_LA"
      if(jiand(mode,mask).eq.2) print *,"VML_HA"

      mask=VML_FPUMODE_MASK
      if(jiand(mode,mask).eq.0) print *,"VML_DEFAULT_PRECISION"
      if(jiand(mode,mask).eq.16) print *,"VML_FLOAT_CONSISTENT "
      if(jiand(mode,mask).eq.32) print *,"VML_DOUBLE_CONSISTENT"
      if(jiand(mode,mask).eq.48) print *,"VML_RESTORE"

      mask=VML_ERRMODE_MASK
      if(jiand(mode,VML_ERRMODE_IGNORE).eq.VML_ERRMODE_IGNORE)          &
     &  print *,"VML_ERRMODE_IGNORE"
      if(jiand(mode,VML_ERRMODE_ERRNO).eq.VML_ERRMODE_ERRNO)            &
     &  print *,"VML_ERRMODE_ERRNO"
      if(jiand(mode,VML_ERRMODE_STDERR).eq.VML_ERRMODE_STDERR)          &
     &  print *,"VML_ERRMODE_STDERR"
      if(jiand(mode,VML_ERRMODE_EXCEPT).eq.VML_ERRMODE_EXCEPT)          &
     &  print *,"VML_ERRMODE_EXCEPT"
      if(jiand(mode,VML_ERRMODE_CALLBACK).eq.VML_ERRMODE_CALLBACK)      &
     &  print *,"VML_ERRMODE_CALLBACK"

      end

File: 10.0.3.020/examples/vmlc/source/mode.c
/*******************************************************************************
!                             INTEL CONFIDENTIAL
!  Copyright(C) 2001-2008 Intel Corporation. All Rights Reserved.
!  The source code contained  or  described herein and all documents related to
!  the source code ("Material") are owned by Intel Corporation or its suppliers
!  or licensors.  Title to the  Material remains with  Intel Corporation or its
!  suppliers and licensors. The Material contains trade secrets and proprietary
!  and  confidential  information of  Intel or its suppliers and licensors. The
!  Material  is  protected  by  worldwide  copyright  and trade secret laws and
!  treaty  provisions. No part of the Material may be used, copied, reproduced,
!  modified, published, uploaded, posted, transmitted, distributed or disclosed
!  in any way without Intel's prior express written permission.
!  No license  under any  patent, copyright, trade secret or other intellectual
!  property right is granted to or conferred upon you by disclosure or delivery
!  of the Materials,  either expressly, by implication, inducement, estoppel or
!  otherwise.  Any  license  under  such  intellectual property  rights must be
!  express and approved by Intel in writing.
!
!*******************************************************************************
!  Content:
!    vmlSetMode/vmlGetMode  Example Program Text
!******************************************************************************/

#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

void PrintTextVmlMode(unsigned int );

int main()
{
  unsigned MKL_INT mode;

  printf("vmlSetMode/vmlGetMode example program\n\n");

  mode=(unsigned MKL_INT)vmlGetMode();
  printf("Default value of vmlMode: ");
  PrintTextVmlMode((unsigned int)mode);
  printf(" (0x%x)\n",(unsigned int)mode);

  mode=(unsigned MKL_INT)(VML_LA|VML_FLOAT_CONSISTENT|VML_ERRMODE_IGNORE);
  vmlSetMode(mode);
  mode=(unsigned MKL_INT)vmlGetMode();
  printf("Value of vmlMode after using vmlSetMode: ");
  PrintTextVmlMode((unsigned int)mode);
  printf(" (0x%x)\n",(unsigned int)mode);

  return 0;
}

void PrintTextVmlMode(unsigned int mode)
{
  switch(mode&VML_ACCURACY_MASK) {
    case 0x001: {
                printf("VML_LA");
                break;
    }
    case 0x002: {
                printf("VML_HA");
                break;
    }
  }
  switch(mode&VML_FPUMODE_MASK) {
    case 0x000: {
                printf(" | VML_DEFAULT_PRECISION");
                break;
    }
    case 0x010: {
                printf(" | VML_FLOAT_CONSISTENT");
                break;
    }
    case 0x020: {
                printf(" | VML_DOUBLE_CONSISTENT");
                break;
    }
    case 0x030: {
                printf(" | VML_RESTORE");
                break;
    }
  }
  if(mode&VML_ERRMODE_IGNORE) {
    printf(" | VML_ERRMODE_IGNORE");
  }
  else {
    if(mode&VML_ERRMODE_ERRNO) {
      printf(" | VML_ERRMODE_ERRNO");
    }
    if(mode&VML_ERRMODE_STDERR) {
      printf(" | VML_ERRMODE_STDERR");
    }
    if(mode&VML_ERRMODE_EXCEPT) {
      printf(" | VML_ERRMODE_EXCEPT");
    }
    if(mode&VML_ERRMODE_CALLBACK) {
      printf(" | VML_ERRMODE_CALLBACK");
    }
  }
}
